benchmark/mnist_benchmark_visualiser.py

```diff
@@ -0,0 +1,167 @@
+        print(f"Error loading the file: {e}")
+        return {}
+
+
+def compute_statistics(data_by_solver: dict, coreset_sizes: list[int]) -> tuple[dict]:
+    """
+    Compute statistical summaries (mean, minimum, and maximum).
+
+    :param data_by_solver: A dictionary where each key is an algorithm name,
+                           and each value is a dictionary mapping coreset size
+                           to benchmark results. Benchmark results include multiple
+                           runs with 'accuracy' and 'time_taken'.
+    :param coreset_sizes: A list of integer coreset sizes to evaluate.
+    :return: A tuple containing two dictionaries:
+             - The first dictionary maps each algorithm name to its accuracy statistics,
+               with keys 'means', 'mins', and 'maxs'.
+             - The second dictionary maps each algorithm name to its time statistics,
+               also with keys 'means', 'mins', and 'maxs'.
+    """
+    accuracy_stats = {
+        algo: {"means": [], "mins": [], "maxs": []} for algo in data_by_solver
+    }
+    time_stats = {
+        algo: {"means": [], "mins": [], "maxs": []} for algo in data_by_solver
+    }
+
+    for algo, sizes in data_by_solver.items():
+        for size in coreset_sizes:
+            size_str = str(size)
+            accuracies, times = [], []
+            if size_str in sizes:
+                for run_data in sizes[size_str].values():
+                    accuracies.append(run_data["accuracy"])
+                    times.append(run_data["time_taken"])
+
+            accuracy_stats[algo]["means"].append(np.mean(accuracies))
+            accuracy_stats[algo]["mins"].append(np.min(accuracies))
+            accuracy_stats[algo]["maxs"].append(np.max(accuracies))
+            time_stats[algo]["means"].append(np.mean(times))
+            time_stats[algo]["mins"].append(np.min(times))
+            time_stats[algo]["maxs"].append(np.max(times))
+
+    return accuracy_stats, time_stats
+
+
+def plot_performance(
+    index: list[float],
+    bar_width: float,
+    data_dict: dict[str, dict[str, list[float]]],
+    coreset_sizes: list[int],
+    ylabel: str,
+    title: str,
+    log_scale: bool = True,
+) -> None:
+    """
+    Plot performance statistics for each algorithm over varying coreset sizes.
+
+    :param index: Array of positions for bar locations.
+    :param bar_width: Width of each bar in the plot.
+    :param data_dict: Dictionary of data statistics for each algorithm.
+    :param coreset_sizes: List of coreset sizes to display on the x-axis.
+    :param ylabel: Label for the y-axis.
+    :param title: Title of the plot.
+    :param log_scale: Option to use a logarithmic scale for the y-axis.
+    """
+    for i, algo in enumerate(data_dict):
+        plt.bar(
+            index + i * bar_width,
+            data_dict[algo]["means"],
+            bar_width,
+            label=f"{algo} ({ylabel})",
+            yerr=[
```

Comment:
using min-max range as error bars isn't ideal. error bars typically represent standard error or confidence intervals, so using them for min-max range could be confusing.

Consider overlaying the individual points - a bar-dot plot - which might give the most accurate picture of results without implying more statistical confidence than we have